### ISSUE TITLE
Skewed-wake correction

### ISSUE THREAD

**Issue by MiguelMarante** [WISDEM/CCBlade#30]
Hello,

I have been working with CCBlade for the past few months, and I only added Pitt Peters Skewed wake correction to the inductionFactors Fortran subroutine:

![image](https://github.com/WISDEM/CCBlade/assets/133772056/0525c359-3f90-4e5f-bcc8-1687b2210af5)

However, I have been struggling to apply this correction, since there are no solutions for certain conditions (high yaw angle, high local tip speed ratio, and downwind positions (azimuth between 90 and 180 degrees)) and what I am working on demands that I test these conditions.

I have come up with different ways to come up with approximations for these conditions, but none of them work well.

I know that this is probably not a suitable question for this forum, but I was wondering if you have suggestions to this problem, since you might have face this also in your work.

Best regards,
Miguel

**Comment by gbarter** [comment]
@andrewning - Maybe you have some insight here?

**Comment by andrewning** [comment]
Not sure I have anything particularly insightful to add but I’ll try. I was going to send you my paper where we explored several skewed wake models but I see that the screenshot is from that paper. My recollection is that the Pitts-Peters model, while relatively easy to apply, was not particular accurate. I’m not surprised that there are cases that struggle to solve. The coupled method of the next section is more complex but performed better. Not sure however if it will be helpful for the conditions you are after. 

For the Pitts-Peters case how do you know there is no solution?  Did you sweep phi?  It could be that their default bracketing is missing the solution. This version of CCBlade isn’t as robust in searching phi for really off-design cases as is my more recent version.

**Comment by MiguelMarante** [comment]
Hello Dr. Ning,

I found out that there is no solution by looping phi from 0 to pi/2 and plotting the curve for the residue. 

Considering the NREL turbine, the following figure shows f0 as a function of phi (the solution is found where the function crosses the x axis) for an azimuth of 90 degrees (0º = 12 o’clock) and a radial position of r = 58.9m (NACA6_A17).


![image](https://github.com/WISDEM/CCBlade/assets/133772056/2fdbac6b-2dbe-4a0e-885e-2a06d8612e73)


It can be observed that for higher yaw angles there are no solutions to the equations. Similar conclusions can be drawned from increasing the local tip speed ratio, or moving from upwind positions to downwind positions.

By increasing the empirical constant, this problem arises for even smaller yaw angles and I think you mentioned it in your paper when you said that constants larger than 1, cause the induction factor to become larger than 1. I think this happens because there ceases to exist feasible solutions (a in [0,1] or phi in [0,pi/2]) for the set of equations.

On other thing that might lead to this problem is that I applied the skewed wake correction locally, since it seems to me that is how it is applied in aerodyn:

![image](https://github.com/WISDEM/CCBlade/assets/133772056/eeb2b760-5445-4e90-9f19-870e6b9a7cb1)


It doesn't seem to me that it has to do with the search for phi using Brent's method, but rather with the fact that there are actually no roots.

Thank you so much for you attention and for your time.

Best regards,
Miguel Pereira

**Comment by andrewning** [comment]
Yeah that makes sense. It’s been a long time since I looked at this but that sounds familiar. It’s just a simple correction so at some point it does lead to inconsistencies with the induction factor.  More generally once the yaw angle becomes large the physical behavior is very different so you’ll have a hard time getting meaningful results from a basic momentum model with a “correction”. 

I guess what I meant by the more robust version is that it is more careful about allowing all directions of inflow in a consistent manner. 

Adding the coupled method may allow you to push it a bit farther, but again a momentum method won’t really be applicable at some point.
